### PR TITLE
Refactor looking up containers in pkg/domain/infra abi

### PR DESCRIPTION
**Pull request by vrothberg**
While analyzing https://github.com/containers/podman/issues/15367 I read a lot of the code in pkg/domain/infra/abi. I found it to be quite hard to read and saw lot's of redundancy. This PR refactors looking up containers in a way that it's easy to add new features, and it paves the way to add the "ignore" semantics there directly.

Please refer to the individual commit messages for details.

```release-note
None
```

### CODE REVIEW

File: pkg/domain/infra/abi/containers.go [hunk 1]
```diff
@@ -1803,16 +1766,16 @@ func (ic *ContainerEngine) ContainerUpdate(ctx context.Context, updateOptions *e
 	if err != nil {
 		return "", err
 	}
-	ctrs, err := getContainersByContext(false, false, false, []string{updateOptions.NameOrID}, ic.Libpod)
+	containers, err := getContainers(ic.Libpod, getContainersOptions{names: []string{updateOptions.NameOrID}})
```
Comment (Luap99) [inline pkg/domain/infra/abi/containers.go:1769]: why is this calling getContainers() at all? If it only need one contianer without latest option it should just use LookupContainer(),

also 
```
if len(containers) != 1 {
		return "", fmt.Errorf("container not found")
}
```
is just a misleading error

Comment (vrothberg) [reply pkg/domain/infra/abi/containers.go:1769]: Fair point.  Cool if I tackle that in another PR?  There is a lot to improve in this file and this code in question was pre-dates this PR.

My main goal at the moment is to improve the `podman rm` code in hope to get closer to resolving https://github.com/containers/podman/issues/15367.

Comment (Luap99) [reply pkg/domain/infra/abi/containers.go:1769]: sure

**Review by Luap99 — COMMENTED** [review]
LGTM

**Pull request merged**